packages/@react-native-windows/cli/src/codegen.ts

```diff
@@ -0,0 +1,298 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ * @format
+ */
+
+import path from 'path';
+import chalk from 'chalk';
+import {performance} from 'perf_hooks';
+
+import {
+  newSpinner,
+  setExitProcessWithError,
+} from './runWindows/utils/commandWithProgress';
+
+import {
+  Command,
+  CommandOption,
+  Config,
+} from '@react-native-community/cli-types';
+import {Telemetry, CodedError} from '@react-native-windows/telemetry';
+import {
+  getDefaultOptions,
+  startTelemetrySession,
+  endTelemetrySession,
+} from './runWindows/utils/telemetryHelpers';
+
+import {runCodeGen} from '@react-native-windows/codegen';
+import {Ora} from 'ora';
+
+export class CodeGenWindows {
+  private changesNecessary: boolean;
+
+  public areChangesNeeded() {
+    return this.changesNecessary;
+  }
+
+  constructor(readonly root: string, readonly options: CodeGenOptions) {
+    this.changesNecessary = false;
+  }
+
+  public async run(spinner: Ora) {
+    const verbose = this.options.logging;
+
+    verboseMessage('', verbose);
+
+    verboseMessage('Loading codegenConfig from package.json');
+    const pkgJson = require(path.join(this.root, 'package.json'));
+
+    if (!pkgJson.codegenConfig) {
+      spinner.info(
+        `No ${chalk.bold(
+          'codegenConfig',
+        )} specified in package.json - ${chalk.yellow(
+          'Skipping codegen-windows',
+        )}`,
+      );
+      return;
+    }
+
+    const codegenConfigType = pkgJson.codegenConfig.type;
+    if (codegenConfigType !== 'modules' && codegenConfigType !== 'all') {
+      spinner.info(
+        `${chalk.bold(
+          'codegenConfig.type',
+        )} in package.json is not ${chalk.bold('modules')} or ${chalk.bold(
+          'all',
+        )} - ${chalk.yellow('Skipping codegen-windows')}`,
+      );
+      return;
+    }
+
+    if (!pkgJson.codegenConfig.windows) {
+      spinner.info(
+        `No ${chalk.bold(
+          'codegenConfig.windows',
+        )} specified in package.json - ${chalk.yellow(
+          'Skipping codegen-windows',
+        )}`,
+      );
+      return;
+    }
+
+    if (!pkgJson.codegenConfig.windows.namespace) {
+      throw new CodedError(
+        'InvalidCodegenConfig',
+        `Missing ${chalk.bold(
```

Comment:
>${chalk.bold( [](http://example.com/codeflow?start=17&length=13)

nit. just checking if okay to have TTY command control characters in an error object...